sc_demographics.R

```diff
@@ -0,0 +1,280 @@
+  )
+
+
+## clean up data ##
+sc_demog <- sc_demog %>%
+  mutate(
+    # use chi if upi is NA
+    upi = if_else(is.na(upi), chi_upi, upi),
+    # check gender code - replace code 99 with 9
+    submitted_gender = replace(submitted_gender, submitted_gender == 99, 9)
+  ) %>%
+  mutate(
+    # use chi gender if avaliable
+    gender = if_else(is.na(chi_gender_code) | chi_gender_code == 9, submitted_gender, chi_gender_code),
+    # use chi dob if avaliable
+    dob = if_else(is.na(chi_date_of_birth), submitted_date_of_birth, chi_date_of_birth)
+  )
+
+
+## postcode ##
+# clean-up
+sc_demog <- sc_demog %>%
+  mutate(
+    # remove any postcodes with length < 5
+    submitted_postcode = na_if(submitted_postcode, str_length(submitted_postcode) < 5 | str_length(submitted_postcode) == 5),
+    chi_postcode = na_if(chi_postcode, str_length(chi_postcode) < 5 | str_length(chi_postcode) == 5)
+  ) %>%
+  mutate(
+    # remove spaces
+    submitted_postcode = str_replace_all(submitted_postcode, fixed(" "), ""),
+    chi_postcode = str_replace_all(chi_postcode, fixed(" "), "")
+  ) %>%
+  mutate(
+    # remove postcodes with length 8 or longer
+    submitted_postcode = na_if(submitted_postcode, str_length(submitted_postcode) > 8 | str_length(submitted_postcode) == 8),
+    chi_postcode = na_if(chi_postcode, str_length(chi_postcode) > 8 | str_length(chi_postcode) == 8)
+  )
+
+sc_demog <-
+  sc_demog %>%
+  mutate(
+    # add space to create length 7 postcode
+    # length 5 postcode
+    # submitted postcode
+    submitted_postcode = if_else(str_length(submitted_postcode) == 5 & !is.na(submitted_postcode),
+      paste0(substr(submitted_postcode, 1, 2), " ", substr(submitted_postcode, 3, 5)),
+      submitted_postcode
+    ),
+    # chi postcode
+    chi_postcode = if_else(str_length(chi_postcode) == 5 & !is.na(chi_postcode),
+      paste0(substr(chi_postcode, 1, 2), " ", substr(chi_postcode, 3, 5)),
+      chi_postcode
+    )
+  ) %>%
+  mutate(
+    # length 6 postcode
+    # submitted postcode
+    submitted_postcode = if_else(str_length(submitted_postcode) == 6 & !is.na(submitted_postcode),
+      paste0(substr(submitted_postcode, 1, 3), " ", substr(submitted_postcode, 4, 6)),
+      submitted_postcode
+    ),
+    # chi postcode
+    chi_postcode = if_else(str_length(chi_postcode) == 6 & !is.na(chi_postcode),
+      paste0(substr(chi_postcode, 1, 3), " ", substr(chi_postcode, 4, 6)),
+      chi_postcode
+    )
+  )
+
+
+# invalid postcodes
+invalid_postcodes <- c(
+  "DY103DJ",
+  "EH191TR",
+  "EH292EZ",
+  "EH33TNZ",
+  "G46 2NF",
+  "G46 6FY",
+  "G69 2YB",
+  "G73 8NZ",
+  "G74 7SN",
+  "G75 1ZZ",
+  "G75 6DF",
+  "G77 3GT",
+  "G78 6BU",
+  "G78 ITE",
+  "G79 8AJ",
+  "IV178ED",
+  "KA113FP",
+  "KA28OBE",
+  "KA5 1LQ",
+  "KA71JUJ",
+  "KA9 9FG",
+  "KW152SE",
+  "KY1 3DO",
+  "KY13OPX",
+  "L15 0PR",
+  "M16 0GS",
+  "ML3 0GS",
+  "ML7 6AQ",
+  "NK1 1AA",
+  "NK1 1AB",
+  "PA438JP",
+  "PA494JS",
+  "PR2 5AL",
+  "TD8 8JD"
+)
```

Comment:
We can hopefully do without this list through more automated checks.

Reply:
@Moohan will these postcodes be removed through the use of the regex? / remove this part?

Reply:
Can you check if they are, please? When I originally wrote the code, these were postcodes that I'd verified weren't real but still had multiple occurrences in the data. In SPSS I couldn't think of a better way to clean them up anymore but in R we have lots more options, for example testing against the regex.

Clearly it's quite a random list of (not)-postcodes so ideally we wouldn't have to use it and would clean-up/remove those postcodes through more automated/robust methods!

Reply:
found 3 that still make it through the regex check - have included removal of them within line which removes dummy postcode